### PR TITLE
CI: Compile openpilot on macOS

### LINKED ISSUE #32885: Compiling openpilot on macOS
We recently brought back some support for installing most dependencies on macOS (see https://github.com/commaai/openpilot/pull/32819). 
The current test just invokes `scons -h` in order to verify that we have the correct dependencies but it does not actually build anything. We want to continue with the next step: compiling openpilot.

The goal:
- compile openpilot on an arm64 macOS machine.  (You don't need an arm64 macOS machine for doing this bounty since everything will be validated by our CI runner.)

How to do it:
- modify the [current build_mac test](https://github.com/commaai/openpilot/blob/a312a102aa05cd5759912de612fc0adb0ae25c19/.github/workflows/selfdrive_tests.yaml#L89) such that it fully invokes `scons` instead of just calling `scons -h`
- make the test pass

### PR DESCRIPTION
**Pull request by bongbui321**
resolves: [#32885](https://github.com/commaai/openpilot/issues/32885)

depends on: https://github.com/commaai/opendbc/pull/1059

1. enabled build in CI
2. added `gcc@13`
3. fixed typo in `replay/util.c`

### CODE REVIEW

**Comment by bongbui321** [comment]
@maxime-desroches updated. Also probably need to fix the permission issue for the new `build doc`

**Comment by maxime-desroches** [comment]
Nice! Can you add a scons cache to make the build faster? You can look how it is done here for our custom ubuntu20.04 cache: https://github.com/commaai/openpilot/blob/master/.github/workflows/tools_tests.yaml#L65

File: .github/workflows/selfdrive_tests.yaml [hunk 1]
```diff
@@ -106,8 +106,17 @@ jobs:
...
+    - name: Getting scons cache
+      uses: 'actions/cache@v4'
+      with:
+        path: /tmp/scons_cache
+        key: scons-${{ runner.arch }}-macos-${{ env.CACHE_COMMIT_DATE }}-${{ github.sha }}
+        restore-keys: |
+          scons-${{ runner.arch }}-macos-${{ env.CACHE_COMMIT_DATE }}
+          scons-${{ runner.arch }}-macos
+    - name: Building openpilot
+      run: uv run scons -u -j$(nproc)
```
Comment (maxime-desroches) [inline .github/workflows/selfdrive_tests.yaml]:
```suggestion
      run: . .venv/bin/activate && scons -j$(nproc)
```

**Comment by maxime-desroches** [comment]
You also need to save the cache: https://github.com/commaai/openpilot/blob/master/.github/workflows/tools_tests.yaml#L76

**Comment by bongbui321** [comment]
It's redundant I believe, @cache already automatically saves the cache already.

**Comment by maxime-desroches** [comment]
Feel free to open a pr to clean up the caching in CI. The caching is normally done in setup --> auto-cache when using the docker image, but not when doing these non docker builds like ubuntu20.04 and macOS

**Pull request merged**